src/twisted/internet/endpoints.py

```diff
@@ -1466,6 +1467,10 @@ def _parseSSL(
         dhParameters=dhParameters,
         **kw,
     )
+
+    if cipher:
+        if isinstance(cipher, str):
+            cf.getContext().set_cipher_list(str.encode(cipher.replace(",", ":")))
```

Comment:
I think is best to have explicit `ascii` encoding here.

Reply:
[#692e540f](https://github.com/twisted/twisted/pull/12137/commits/692e540f85cac1146fb0a44a230138ca743ce5b0)